l1-contracts/src/core/libraries/Decoder.sol

```diff
@@ -468,40 +474,49 @@ library Decoder {
       offset := add(_offset, 0x4)
     }
 
-    bytes32 logsHash;
-    uint256 iterationLogsLength;
-    uint256 tempLength;
+    bytes memory logsHashes = new bytes(0x40); // A memory to which we will write the 2 logs hashes to be accumulated
```

Comment:
Is the reason this is not using `bytes32[]` that it made sha256 easier later on? Otherwise seemed move intuitive to me to use `bytes32[]` of length 2 where you can write to them directly with `logHash[0] = kernelPublicInputsLogsHash` later on as well.

Reply:
Replacing bytes with bytes32[2] is not problematic but it requires calling converting it to bytes with ` `abi.encodePacked` before hashing it. This raised gas cost of `testComputeKernelLogs2Iterations` from 13014 to 13917 gas.

If we want to optimize for readability now then I would say it's worth it.

Reply:
Addressed in [5e462b9](https://github.com/AztecProtocol/aztec-packages/pull/764/commits/5e462b94ad3019e7cd744d6e5780c1ce2fe1335e)